Gemfile

```diff
@@ -23,3 +21,8 @@ gem 'gherkin', path: ENV['GHERKIN_RUBY'] if ENV['GHERKIN_RUBY']
 gem 'cucumber-messages', path: ENV['CUCUMBER_MESSAGES_RUBY'] if ENV['CUCUMBER_MESSAGES_RUBY']
 
 gem 'cucumber-formatter-dots', path: ENV['CUCUMBER_FORMATTER_DOTS'] if ENV['CUCUMBER_FORMATTER_DOTS']
+
+# Use an older protobuf on JRuby and MRI < 2.5
+gem 'google-protobuf', '~> 3.2.0.2' if RbConfig::CONFIG['MAJOR'].to_i == 2 && RbConfig::CONFIG['MINOR'].to_i < 5 || RUBY_PLATFORM == 'java'
```

Comment:
This can become much simpler and be just re-added to the gemspec at `~> 3.8` once it is cut and released. They've got it working on the current master for 2.3+ (I'm just double checking they've got JRuby working also)